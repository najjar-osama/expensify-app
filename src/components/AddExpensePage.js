import React from "react";
import ExpensForm from "./ExpenseForm";
import { connect } from "react-redux";
import { startAddExpense } from "../actions/expenses";

export class AddExpensePage extends React.Component {
  onSubmit = expense => {
    this.props.startAddExpense(expense);
    this.props.history.push("/");
  };
  render() {
    return (
      <div>
        <h2>Add Expense Page!</h2>
        <ExpensForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startAddExpense: expense => dispatch(startAddExpense(expense))
});
export default connect(
  undefined,
  mapDispatchToProps
)(AddExpensePage);
