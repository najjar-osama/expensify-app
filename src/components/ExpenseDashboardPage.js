import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";
const ExpenseDashboardPage = () => (
  <header>
    <h1>Expensify!</h1>
    <ExpenseListFilters />
    <ExpenseList />
  </header>
);

export default ExpenseDashboardPage;
