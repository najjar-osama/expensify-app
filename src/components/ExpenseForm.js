import React from "react";
import moment from "moment";
import { SingleDatePicker } from "react-dates";

class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      description: props.expense ? props.expense.description : "",
      note: props.expense ? props.expense.note : "",
      amount: props.expense ? (props.expense.amount / 100).toString() : "",
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      calendarFoucesd: false,
      error: ""
    };
  }

  onDescriptionChange = e => {
    const description = e.target.value;
    this.setState(Object.assign({}, this.state, { description }));
  };
  onNoteChange = e => {
    const note = e.target.value;
    this.setState(Object.assign({}, this.state, { note }));
  };
  onAmountChange = e => {
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(Object.assign({}, this.state, { amount }));
    }
  };
  onDateChange = createdAt => {
    if (createdAt) {
      this.setState(Object.assign({}, this.state, { createdAt }));
    }
  };
  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calendarFoucesd: focused }));
  };
  onSubmit = e => {
    e.preventDefault();
    if (!this.state.description || !this.state.amount) {
      const error = "description and amount fields are mandatory.";
      this.setState(Object.assign({}, this.state, { error }));
    } else {
      const error = "";
      this.setState(Object.assign({}, this.state, { error }));
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };

  render() {
    return (
      <div>
        <div>{this.state.error && <p>{this.state.error}</p>}</div>
        <form onSubmit={this.onSubmit}>
          <input
            type="text"
            placeholder="Description"
            value={this.state.description}
            onChange={this.onDescriptionChange}
            autoFocus
          />
          <input
            type="text"
            placeholder="Amount"
            value={this.state.amount}
            onChange={this.onAmountChange}
          />
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFoucesd}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
            id="datePicker_AddExpense"
          />
          <textarea
            placeholder="Add note for your expense (optional)"
            value={this.state.note}
            onChange={this.onNoteChange}
          />
          <button>
            {this.props.expense ? "Update expense" : "Add Expense"}
          </button>
        </form>
      </div>
    );
  }
}

export default ExpenseForm;
