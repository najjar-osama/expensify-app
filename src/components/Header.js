import React from "react";
import { NavLink } from "react-router-dom";
import { startLogout } from "../actions/auth";
import { connect } from "react-redux";
export const Header = ({ startLogout }) => (
  <header>
    <h1>Expensify!</h1>
    <div>
      <NavLink to="/" activeClassName="is-active" exact={true}>
        Home
      </NavLink>
      <NavLink to="/create" activeClassName="is-active">
        Add Expense
      </NavLink>
    </div>
    <button onClick={startLogout}>Log out</button>
  </header>
);

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout())
});
export default connect(
  undefined,
  mapDispatchToProps
)(Header);
