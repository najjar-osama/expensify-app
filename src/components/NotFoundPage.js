import React from "react";
import { Link } from "react-router-dom";

const NotFoundPage = () => (
  <div>
    <h2>404 Not Found</h2>
  </div>
);

export default NotFoundPage;
