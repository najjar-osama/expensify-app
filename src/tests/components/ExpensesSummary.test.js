import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("Should correctly render expenses summary with 1 expense", () => {
  const wrapper = shallow(
    <ExpensesSummary expensesCount={1} expensesTotal={235} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("Should correctly render expenses summary multiple  expenses", () => {
  const wrapper = shallow(
    <ExpensesSummary expensesCount={34} expensesTotal={2356489465442} />
  );
  expect(wrapper).toMatchSnapshot();
});
