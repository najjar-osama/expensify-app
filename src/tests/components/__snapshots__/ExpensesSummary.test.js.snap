// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render expenses summary multiple  expenses 1`] = `
<div>
  <h1>
    Viewing 
    34
     
    expenses
     totaling 
    $23,564,894,654.42
  </h1>
</div>
`;

exports[`Should correctly render expenses summary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $2.35
  </h1>
</div>
`;
